File: src/main/java/project/app/flutter_spring_todoapp/todo/domain/TodoStatus.java
package project.app.flutter_spring_todoapp.todo.domain;

import com.fasterxml.jackson.annotation.JsonCreator;
import lombok.Getter;
import lombok.RequiredArgsConstructor;

import java.util.Arrays;
import java.util.Map;
import java.util.stream.Collectors;

@RequiredArgsConstructor
public enum TodoStatus {

    TODO("todo"),
    IN_PROGRESS("in_progress"),
    DONE("done");

    private final String name;
    private final static Map<String,TodoStatus> statusMap = Arrays.stream(TodoStatus.values())
            .collect(Collectors.toMap(TodoStatus::getName,value -> value));

    @JsonCreator
    public static TodoStatus findStatus(final String source) {
        return statusMap.get(source.toLowerCase());
    }

    public String getName() {
        return name;
    }
}

File: todo_app/lib/enum/TodoStatus.dart
enum TodoStatus {

  todo("시작전","todo"),
  inProgress("진행중","in_progress"),
  done("완료", "done");

  final String label;
  final String value;
  static final Map<String, TodoStatus> todoStatusMap =
  Map.fromEntries(TodoStatus.values.map((status) => MapEntry(status.value, status)));

  const TodoStatus (this.label, this.value);

  static TodoStatus fromValue(String value){
    return todoStatusMap[value.toLowerCase()] ?? TodoStatus.todo;
  }
}